src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -392,7 +392,7 @@ internal static bool IsNonAttributedTypeValidForSerialization(Type type)
             else
             {
                 return (type.IsVisible &&
-                    type.GetConstructor(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public, Array.Empty<Type>()) != null);
+                    type.GetConstructor(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public, null, Array.Empty<Type>(), null) != null);
```

Comment:
Is it uncommon to use named parameters in this codebase?

Reply:
No it isn't uncommon, but I'm not sure it adds a ton of value in these cases.

Note I've opened #42753 to make this overload an official API, so these `null`s won't be necessary in the future.